planetscale/imports.go

```diff
@@ -0,0 +1,306 @@
+package planetscale
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type DataImportSource struct {
+	HostName            string `json:"hostname"`
+	Database            string `json:"schema_name"`
+	Port                int    `json:"port"`
+	SSLMode             string `json:"ssl_mode"`
+	SSLVerificationMode ExternalDataSourceSSLVerificationMode
+	UserName            string `json:"username"`
+	Password            string `json:"password"`
+}
+
+type ExternalDataSourceSSLVerificationMode int
+
+const (
+	SSLModeDisabled ExternalDataSourceSSLVerificationMode = iota
+	SSLModePreferred
+	SSLModeRequired
+	SSLModeVerifyCA
+	SSLModeVerifyIdentity
+)
+
+func (sm ExternalDataSourceSSLVerificationMode) String() string {
+	switch sm {
+	case SSLModeDisabled:
+		return "disabled"
+	case SSLModePreferred:
+		return "preferred"
+	case SSLModeRequired:
+		return "required"
+	case SSLModeVerifyCA:
+		return "verify_ca"
+	default:
+		return "verify_identity"
+	}
+}
+
+type DataImportState int
+
+const (
+	DataImportPreparingDataCopy DataImportState = iota
+	DataImportPreparingDataCopyFailed
+	DataImportCopyingData
+	DataImportCopyingDataFailed
+	DataImportSwitchTrafficPending
+	DataImportSwitchTrafficRunning
+	DataImportSwitchTrafficCompleted
+	DataImportSwitchTrafficError
+	DataImportReverseTrafficRunning
+	DataImportReverseTrafficCompleted
+	DataImportReverseTrafficError
+	DataImportDetachExternalDatabaseRunning
+	DataImportDetachExternalDatabaseError
+	DataImportReady
+)
+
+var stateToImportStateMap = map[string]DataImportState{
+	"prepare_data_copy_pending":        DataImportPreparingDataCopy,
+	"prepare_data_copy_error":          DataImportPreparingDataCopyFailed,
+	"data_copy_pending":                DataImportCopyingData,
+	"data_copy_error":                  DataImportCopyingDataFailed,
+	"switch_traffic_workflow_pending":  DataImportSwitchTrafficPending,
+	"switch_traffic_workflow_running":  DataImportSwitchTrafficRunning,
+	"switch_traffic_workflow_error":    DataImportSwitchTrafficError,
+	"reverse_traffic_workflow_running": DataImportReverseTrafficRunning,
+	"reverse_traffic_workflow_error":   DataImportReverseTrafficError,
+	"cleanup_workflow_pending":         DataImportSwitchTrafficCompleted,
+	"cleanup_workflow_running":         DataImportDetachExternalDatabaseRunning,
+	"cleanup_workflow_error":           DataImportDetachExternalDatabaseError,
```

Comment:
These three names don't match up with the variables. I think we should maybe rename the variables in that case, to make sure the string values match the variable names.